quantumdb-postgresql/src/main/java/io/quantumdb/core/versioning/Backend.java

```diff
@@ -153,10 +147,13 @@ public State load(Connection connection, Catalog catalog) throws SQLException {
 		while (!toDo.isEmpty()) {
 			Version version = toDo.remove(0);
 			for (Entry<RefId, String> entry : tableVersions.column(version).entrySet()) {
-				TableRef tableRef = refLog.getTableRefs(version).stream()
-						.filter(ref -> ref.getName().equals(entry.getValue()) && ref.getRefId().equals(entry.getKey().getRefId()))
-						.findFirst()
-						.orElse(null);
+				TableRef tableRef = null;
+				if (version.getParent() != null) {
+					tableRef = refLog.getTableRefs(version.getParent()).stream()
```

Comment:
Without the .getParent() there will be no linking of TableRefs as it will always make a new one. In a previous commit f22500b52e878d1a38826ffa4b587cd70e5158b7 where you refactored TableId to refId the .getParent() got removed.